src/main/java/org/jnode/fs/xfs/directory/BPlusTreeDirectory.java

```diff
@@ -121,4 +99,35 @@ public BPlusTreeDirectory(byte [] data, long offset, long iNodeNumber, XfsFileSy
         final NodeDirectory leafDirectory = new NodeDirectory(buffer.array(), 0, fileSystem, iNodeNumber, extents, leafExtentIndex);
         return leafDirectory.getEntries(parentDirectory);
     }
+
+
+    public List<DataExtent> getFlattenedExtents(byte[] data,int level,List<DataExtent> currentExtents, int btreeBlockOffset, int numrecs) throws IOException {
```

Comment:
the `data` that's being passed in is just the buffer in the parent class, so it seems unnecessary? If it's just using the data from the parent class, the usages of `BigEndian.get*()` in the function looks like it can just be resolved to `XfsObject.getUInt*()`.

Reply:
Because of the recursive nature of the function that kind of fix would require lots of code duplication